tests/suites/verify.js

```diff
@@ -144,6 +144,18 @@ export function verifySuite({
           const credential = cloneTestVector(disclosed?.invalid?.valuePrefix);
           await verificationFail({credential, verifier});
         });
+        it('If the result is not an array of five, six, or seven elements ' +
+          '— a byte array, a map of integers to integers, two arrays of ' +
+          'integers, and one or two byte arrays — an error MUST be raised ' +
```

Comment:
```
  1 byte array
+ 1 map of integers to integers
+ 2 arrays of integers
```
That's 4 elements.
Add `1 or 2 byte arrays`, and that's 5 or 6 elements, total.

What is the possible 7th element?

Reply:
@TallTed this is direct from the latest spec here: https://w3c.github.io/vc-di-bbs/#parsederivedproofvalue

I was under the impression that the 7th element was due to changes in related bbs specs WRT to the key, blind signing, and pseudonym signing, but I'm not aware of exactly why this came in. It looks like those changes were made here: https://github.com/w3c/vc-di-bbs/issues/181

@Wind4Greg do you know what the 7th element refers to in this statement? I believe this statement was merged in because the blind and pseudonym specs changed.

It sounds like this might be an issue for the spec itself though and not the test suite.

Reply:
@TallTed fairly confident the 7th element is: https://github.com/w3c/vc-di-bbs/blob/ad66d386015efc775a0320676381d0af951d6af2/index.html#L679-L686

So related to blind and pseudonym algorithms.

Reply:
The first paragraph of [3.3.7 parseDerivedProofValue](https://w3c.github.io/vc-di-bbs/#parsederivedproofvalue) —

> A single derived proof value object is produced as output, which contains a set of six or seven elements, having the names "bbsProof", "labelMap", "mandatoryIndexes", "selectiveIndexes", "presentationHeader", "featureOption", and, depending on the value of the featureOption parameter, "pseudonym" and/or "lengthBBSMessages".

— is internally inconsistent (the set may have six, seven, or eight elements) and it disagrees with the last paragraph of that algorithm (again, should say "six, seven, or eight elements" including "featureOption", "pseudonym" and/or "lengthBBSMessages") —

> Return derived proof value as an object with properties set to the five, six, or seven elements, using the names "bbsProof", "labelMap", "mandatoryIndexes", "selectiveIndexes", "presentationHeader", and optional "pseudonym" and/or "lengthBBSMessages", respectively. In addition, add featureOption and its value to the object.

These should be brought into agreement. Whatever the result is, it should then be applied to this part of the test suite.

Reply:
Moved to #64 which will probably move to the https://github.com/w3c/vc-di-bbs/ repo.

Reply:
#64 is untouched. The enumeration of the elements (`a byte array, a map of integers to integers, two arrays of integers, and one or two byte arrays`) here does not agree with the possible counts indicated (`five, six, or seven elements`). The enumeration only leads to `five or six elements`; there is no indication of a possible seventh.